neo4j-to-neptune/src/test/java/com/amazonaws/services/neptune/io/Neo4jStreamWriterIntegrationTest.java

```diff
@@ -0,0 +1,195 @@
+import org.junit.Assume;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+
+import static org.junit.Assert.*;
+
+/**
+ * Integration tests for Neo4jStreamWriter class.
+ * These tests require a running Neo4j database with APOC plugin installed.
+ * 
+ * To run these tests:
+ * 1. Start a Neo4j database (e.g., using Docker: docker run -p 7474:7474 -p 7687:7687 -e NEO4J_AUTH=neo4j/password neo4j:latest)
+ * 2. Install APOC plugin
+ * 3. Set system properties: -Dneo4j.uri=bolt://localhost:7687 -Dneo4j.username=neo4j -Dneo4j.password=password
+ * 4. Run tests
+ */
+public class Neo4jStreamWriterIntegrationTest {
+
+    @Rule
+    public TemporaryFolder tempFolder = new TemporaryFolder();
+
+    private Directories directories;
+    private Neo4jStreamWriter writer;
+    
+    // Test configuration - can be overridden with system properties
+    private static final String DEFAULT_URI = "bolt://localhost:7687";
+    private static final String DEFAULT_USERNAME = "neo4j";
+    private static final String DEFAULT_PASSWORD = "password";
+
+    @Before
+    public void setUp() throws IOException {
+        // Create a temporary directory for testing
+        File tempDir = tempFolder.newFolder("integration-test-output");
+        directories = Directories.createFor(tempDir);
+
+        // Get connection details from system properties or use defaults
+        String uri = System.getProperty("neo4j.uri", DEFAULT_URI);
+        String username = System.getProperty("neo4j.username", DEFAULT_USERNAME);
+        String password = System.getProperty("neo4j.password", DEFAULT_PASSWORD);
+
+        try {
+            writer = new Neo4jStreamWriter(uri, username, password, directories);
+
+            createTestData();
+                
+        } catch (Exception e) {
+            Assume.assumeNoException("Neo4j database is not available - skipping integration tests", e);
+        }
+    }
+
+    @After
+    public void tearDown() {
+        if (writer != null) {
+            writer.close();
+        }
+    }
+
+    /**
+     * Helper method to create test data in the Neo4j database
+     */
+    private void createTestData() {
+        // TODO decide on test data set
+    }
+
+    @Test
+    public void testStreamToFileWithRealDatabase() throws IOException {
+        // First, create some test data
+        createTestData();
+
+        // Stream data to file
+        File result = writer.streamToFile("integration-test");
+
+        // Verify results
+        assertNotNull("Result file should not be null", result);
+        assertTrue("Result file should exist", result.exists());
+        assertTrue("Result file should not be empty", result.length() > 0);
+
+        // Read and verify file contents
+        String content = new String(Files.readAllBytes(result.toPath()));
+        assertFalse("File content should not be empty", content.trim().isEmpty());
+        
+        System.out.println("Generated file content preview:");
```

Comment:
Once the test dataset is chosen, it would be good to replace this with assertions that verify the exported data is correct.